lib/anno_utils.pm

```diff
@@ -0,0 +1,2917 @@
+            $oldtype, $types_ref) = $self->_create_inputgenome_from_genome(
+                                                    $inputgenome, $input_obj_ref);
+        $rast_details{oldfunchash} = $oldfunchash;
+        $rast_details{oldtype} = $oldtype;
+        $rast_details{types} = $types_ref;
+    } elsif ($is_assembly) {
+        print "INFO:----$input_obj_ref points to an assembly----\n";
+        ($inputgenome, $contigobj,
+            $contigID_hash) = $self->_create_inputgenome_from_assembly(
+                                                    $inputgenome, $input_obj_ref);
+    }
+
+    $parameters->{genetic_code} = $inputgenome->{genetic_code};
+    $parameters->{domain} = $inputgenome->{domain};
+    $parameters->{scientific_name} = $inputgenome->{scientific_name};
+
+    $rast_details{parameters} = $parameters;
+    $rast_details{contigobj} = $contigobj;
+    $rast_details{contigID_hash} = $contigID_hash;
+    $rast_details{is_genome} = $is_genome;
+    $rast_details{is_assembly} = $is_assembly;
+    return (\%rast_details, $inputgenome);
+}
+
+
+#
+## Recording the annotation process by taking a note in $message;
+## Inserting contigs into $inputgenome
+## Input:
+##  ($rast_details = {
+##      parameters => $parameters,
+##      oldfunchash => $oldfunchash,
+##      oldtype => $oldtype,
+##      types => \%types,
+##      contigobj => $contigobj,
+##      fasta_file => $fasta_file,
+##      gff_file => $gff_file
+##  }, $inputgenome);
+## Returns:
+##  ($rast_details = {
+##      parameters => $parameters,
+##      oldfunchash => $oldfunchash,
+##      oldtype => $oldtype,
+##      types => \%types,
+##      contigobj => $contigobj,
+##      message => $message,
+##      tax_domain => $tax_domain,
+##      fasta_file => $fasta_file,
+##      gff_file => $gff_file
+##  }, $inputgenome);
+#
+sub _set_messageNcontigs {
+    my ($self, $rast_in, $inputgenome) = @_;
+
+    my %rast_details = %{ $rast_in };
+    my $parameters = $rast_details{parameters};
+    my $contigobj = $rast_details{contigobj};
+    my $is_assembly = $rast_details{is_assembly};
+
+    my $tax_domain = 'U';
+    my $message = "";
+    if (defined($inputgenome->{domain})) {
+        $tax_domain = ($inputgenome->{domain} =~ m/^([ABV])/o) ? $inputgenome->{domain} : 'U';
+        if ($tax_domain eq 'U' ) {
+            $message .= "Some RAST tools will not run unless the taxonomic domain is Archaea, Bacteria, or Virus. \nThese tools include: call selenoproteins, call pyrroysoproteins, call crisprs, and call prophage phispy features.\nYou may not get the results you were expecting with your current domain of $inputgenome->{domain}.\n";
+        }
+    }
+    if (defined($contigobj)) {
+        my $count = 0;
+        my $size = 0;
+        if (defined($contigobj->{contigs})) {
+            $inputgenome->{contigs} = $contigobj->{contigs};
+            for (my $i=0; $i < @{$inputgenome->{contigs}}; $i++) {
+                $count++;
```

Comment:
No need for `$i` and `$count`

Reply:
$i might not be needed, but $count is needed for building the $message.